pkg/service/egresstracking/service.go

```diff
@@ -128,5 +144,149 @@ func (s *EgressTrackingService) egressTrack(ctx context.Context, batchCID cid.Ci
+	if s.cleanupCancel != nil {
+		s.cleanupCancel()
+	}
+
+	select {
+	case <-s.cleanupDone:
+		log.Info("cleanup task stopped")
+		return nil
+	case <-ctx.Done():
+		return fmt.Errorf("timeout waiting for cleanup task to stop: %w", ctx.Err())
+	}
+}
+
+func (s *EgressTrackingService) runCleanupTask(ctx context.Context) {
+	defer close(s.cleanupDone)
+
+	ticker := time.NewTicker(s.cleanupCheckInterval)
+
+	for {
+		select {
+		case <-ctx.Done():
+			log.Info("cleanup task context cancelled")
+			return
+		case <-ticker.C:
+			if err := s.cleanupConsolidatedBatches(ctx); err != nil {
+				log.Errorf("error cleaning up consolidated batches: %v", err)
+			}
+		}
+	}
+}
+
+func (s *EgressTrackingService) cleanupConsolidatedBatches(ctx context.Context) error {
+	// List all batches
+	batchCIDs, err := s.journal.List(ctx)
+	if err != nil {
+		return fmt.Errorf("listing batches: %w", err)
+	}
+
+	if len(batchCIDs) == 0 {
+		log.Debug("no batches to clean up")
+		return nil
+	}
+
+	log.Debugf("checking %d batches for consolidation", len(batchCIDs))
+
+	// Check each batch for consolidation
+	for _, batchCID := range batchCIDs {
+		if err := s.checkAndRemoveConsolidatedBatch(ctx, batchCID); err != nil {
+			log.Errorf("error checking batch %s: %v", batchCID, err)
+			// Continue with other batches even if one fails
+		}
+	}
+
+	return nil
+}
+
+func (s *EgressTrackingService) checkAndRemoveConsolidatedBatch(ctx context.Context, batchCID cid.Cid) error {
+	// Get the consolidate invocation CID from the consolidation store
+	consolidateInvCID, err := s.consolidationStore.GetConsolidateInvocationCID(ctx, batchCID)
+	if err != nil {
+		log.Warnf("batch %s not found in consolidation store, skipping: %v", batchCID, err)
+		return nil
+	}
+
+	// Fetch the consolidate receipt from the egress tracker's receipts endpoint
+	rcpt, err := s.rcptsClient.Fetch(ctx, cidlink.Link{Cid: consolidateInvCID})
+	if err != nil {
+		if errors.Is(err, receipts.ErrNotFound) {
+			log.Debugf("consolidate receipt not yet available for batch %s", batchCID)
+			return nil
+		}
+
+		return fmt.Errorf("fetching consolidate receipt: %w", err)
+	}
+
+	log.Debugf("consolidate receipt fetched for batch %s", batchCID.String())
+
+	// Fetch the original track invocation for the batch
+	trackInv, err := s.consolidationStore.GetTrackInvocation(ctx, batchCID)
+	if err != nil {
+		return fmt.Errorf("batch %s not found in consolidation store: %w", batchCID, err)
+	}
+
+	s.validateConsolidateReceipt(rcpt, trackInv)
+
+	// Remove the batch from the store
+	log.Infof("consolidate receipt found for batch %s, removing from store", batchCID)
+	if err := s.journal.Remove(ctx, batchCID); err != nil {
+		return fmt.Errorf("removing consolidated batch: %w", err)
+	}
+
+	// Remove from consolidation store
+	if err := s.consolidationStore.Delete(ctx, batchCID); err != nil {
+		log.Warnf("failed to remove batch %s from consolidation store: %v", batchCID, err)
+	}
```

Comment:
Why return an error for `journal.Remove` but only log for `consolidationStore.Delete`? Just curious.

Reply:
failing to remove the batch from the journal means it will be checked again. OTOH, failing to remove from the consolidation store does no harm beyond the records being kept in the datastore. It's a subtle difference, happy to change the log into an error if you think that makes more sense.